### PR TITLE
Various Tasks

### PR DESCRIPTION
**Pull request by sjmiles**

### CODE REVIEW

File: core/ts/isolation/vanilla.ts [hunk 1]
```diff
@@ -82,6 +82,25 @@ const requireImplFactory = async (kind, options) => {
   return globalThis.harden(factory);
 };
 
+const {assign, keys, entries, values, create} = Object;
+
+globalThis.SafeObject = {
```
Comment (mariakleiner) [inline core/ts/isolation/vanilla.ts:87]: thank you, this is fantastic!

File: pkg/Library/NodeGraph/Runner.js [hunk 2]
```diff
@@ -164,8 +164,8 @@ encodeFullNodeKey({key}, {$meta}) {
 updateNodeInPipeline(node, pipeline) {
   const index = pipeline?.nodes?.findIndex(n => n.key === node.key);
   // TODO (b/245770204): avoid copying objects
-  // pipeline.nodes[index] = node;
-  pipeline.nodes = assign([], pipeline.nodes, {[index]: node});
+  pipeline.nodes[index] = node;
```
Comment (mariakleiner) [inline pkg/Library/NodeGraph/Runner.js:167]: i have noticed weirdness that occasionally happens when this is done. this is as much as I know...
i think we might want to be cautious and do this change in a separate PR, so it's easily revertable, in case it breaks rapsai (the NodeTree change won't, but Runner will, if...)
or maybe we hold off with this change until i have more time to investigate, WDYT?

**Review by mariakleiner — APPROVED** [review]

File: core/ts/isolation/vanilla.ts (hunk 1 above)
Comment (sjmiles) [reply core/ts/isolation/vanilla.ts:87]: Glad you spotted it!

File: pkg/Library/NodeGraph/Runner.js (hunk 2 above)
Comment (sjmiles) [reply pkg/Library/NodeGraph/Runner.js:167]: These concepts should be well understood. 

The code is highly biased towards doing nothing if nothing has happened, we generally want to work in concert with that.
